paasta_tools/spark_tools.py

```diff
@@ -264,7 +264,21 @@ def get_spark_memory_in_unit(mem: str, unit: Literal["k", "m", "g", "t"]) -> flo
         try:
             memory_bytes = float(mem[:-1]) * MEM_MULTIPLIER[mem[-1]]
         except (ValueError, IndexError):
-            print(f"Unable to parse memory value {mem}. Defaulting to 2 GB.")
-            memory_bytes = 2147483648  # default to 2 GB
+            print(f"Unable to parse memory value {mem}. Defaulting to 0.")
+            memory_bytes = 0  # default to 0
```

Comment:
Should we just throw an exception to catch the problem before the problem config is pushed / the job is deployed?